### DISCUSSION TITLE
Automatic numbering of duplicate files

### DISCUSSION THREAD

**Discussion by dxmahx** [deniscerri/ytdlnis#444] [Q&A] [answered]
Hello guys. Unfortunately, I'm not that familiar with this app and I've already tried to find something about my problem on the Internet. Unfortunately without success.I would like to download a music mix from YouTube, which is already divided into different chapters.However, the chapters were all named the same, which means that after the download I only see one file.Is there a way that the app still downloads all chapters separately ?

**Comment by zaednasr** [comment]
Split by chapters chip didnt work? 
![image](https://github.com/deniscerri/ytdlnis/assets/75589932/f2509603-a468-4600-8d86-a4b416c7caa2)

**Reply by dxmahx** [reply]
Works, but not if the chapters all have the same name. At the end of the download there is only one audio file.
btw thanks for the quick reply

**Reply by zaednasr** [reply]
Add this as an extra command

 -o "chapter:[%(section_number)s] %(section_title)s.%(ext)s"

**Reply by dxmahx** [reply]
I get the following error message 


 -N "3" --download-archive "/data/user/0/com.deniscerri.ytdl/cache/download_archive.txt" -P "/data/user/0/com.deniscerri.ytdl/cache/downloads/26" o "chapter:[%(section_number)s] % (section_title)s.%(ext)s" "https://youtube.com/watch?v=J0y6wM0aAgE"

ERROR: [generic] 'o' is not a valid URL. Set --default-search "ytsearch" (or run  yt-dlp "ytsearch:o" ) to search YouTube
ERROR: Unable to handle request: Unsupported url scheme: "chapter" (urllib)

**Reply by zaednasr** [reply]
you forgot the - in -o

**Reply by dxmahx** [reply]
When I start the download with this command, I immediately receive a message that the file was instantly downloaded. However, nothing is downloaded

**Reply by zaednasr** [reply]
In the extra commands page, copy that code and then put in the command tab and then add the command i sent you in the end. I guess it should work there. The files will be in the command folder

**Reply by dxmahx** [reply]
![Screenshot_20240331-224017](https://github.com/deniscerri/ytdlnis/assets/122187962/9f3042a6-87a3-4ddb-9b7b-3e83bff093cb)
I'm probably being a bit silly, but do you mean this menu?
![Screenshot_20240331-225835](https://github.com/deniscerri/ytdlnis/assets/122187962/d8b63384-d44f-42e8-a753-eb7311f20c8f)
Do I have to set anything? If I select audio under "Extra command", only the mp4 file is downloaded anyway. Sorry if I'm being silly and thank you for your patience

**Reply by zaednasr** [reply]
nonono

take whatever the audio tab command has and the add the -o thing

**Comment by dxmahx** [comment]
I really don't know what I'm doing wrong. That the download already exists is not correct 

https://github.com/deniscerri/ytdlnis/assets/122187962/ee858a71-7e16-4423-9c88-e717aef8a636

**Reply by zaednasr** [reply, marked as answer]
You are misundestood
Audio, Video, Command 3 separate configs. They are not related

When i meant add extra command. The app has a functionality to add extra commands to the audio , video tab.  You can find the toggle in the processing settings

With it on you can find an add extra commands chip in the audio/video section in the download card
If you tap it you can see the whole command that configuration has

You can copy that and then add the command i gave you and use it in the command tab to see if it works.

https://github.com/deniscerri/ytdlnis/assets/75589932/f5be6b43-dabe-45b2-99be-a64be2d306fc


I removed the url in the command tab because the app puts it there for you by itself

**Reply by dxmahx** [reply]
Now it worked, thank you for your support.Unfortunately, I still have one last "problem" there. That's how the individual chapters are now named for me. What could be the reason?
![Screenshot_20240401-000225](https://github.com/deniscerri/ytdlnis/assets/122187962/6516dc59-47e9-4be2-95c8-e5540fa293fd)

**Reply by zaednasr** [reply]
You have a space after % that's why. It doesn't need a space

**Reply by dxmahx** [reply]
Now everything works fine. Thanks again for your time and patience 🙏🏻
